### ISSUE TITLE
"waitUntil" : "networkidle" for page.screenshot()

### ISSUE THREAD

**Issue by petarthewizard** [puppeteer/puppeteer#799]
It would be very nice to have "waitUntil" : "networkidle" as an option for page.screenshot() and page.pdf() same as in page.goto() so that a screenshot is taken only after the network has been idle for some time.

**Comment by ebidel** [comment]
I guess the request could apply to any of the page methods :)

IMO, since there's already a [couple of ways](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitforselectororfunctionortimeout-options) that you can wait for or delay work, we wouldn't want to overload more methods. `page.goto()` has `waitUntil` because it's tied to navigation / page load and signals like "networkidle" or "load" make sense.

**Comment by Garbee** [comment]
[page.waitForNavigation](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagewaitfornavigationoptions) seems to already fulfill the desired needs. Am I missing something that it isn't allowing for?

**Comment by ebidel** [comment]
Related: https://github.com/GoogleChrome/puppeteer/issues/791

**Comment by petarthewizard** [comment]
I tried page.waitForNavigation and it didn't work but now I think that was because of an error on my part.

**Comment by remorses** [comment]
If you need to wait for `networkidle` without navigation you can implement something like this
```javascript
const waitForLoad = (page) => new Promise((resolve) => {
  page.on('rquest', (req) => {
    waitForLoad(page)
  })
  page.on('requestfinished', (req) => {
    setTimeout(() => resolve("idle"), 800)
  })
})
```
and use like this
```javascript
await waitForLoad(page)
```
